[APP-1031] Add new followerRule to threadgate settings

estrattonbailey opened the pull request:
Adds `followersRule` support to interaction settings. App view and dataplane are already deployed and working, so all that's left is the UI.

![CleanShot 2025-02-06 at 21 53 55@2x](https://github.com/user-attachments/assets/a816b81b-11ea-4863-8984-bb108a824c1b)
![CleanShot 2025-02-06 at 21 53 10@2x](https://github.com/user-attachments/assets/f3f93d70-c65a-4209-99a4-5fa68dbacf8d)
![CleanShot 2025-02-06 at 21 48 40@2x](https://github.com/user-attachments/assets/5f9a0217-18b9-4e5d-a98f-113abc90c41d)
![CleanShot 2025-02-06 at 21 40 30@2x](https://github.com/user-attachments/assets/d86e80a0-9cdb-4d43-9a7a-abc6de2ae532)

mozzius reviewed (approved):
Awesome, works perfectly. Filed #7692 separately.

My only suggestion is maybe disable this enormous CTA when the threadgate is enabled? What do you think? It's probably fine to keep if you disagree

<img width="811" alt="Screenshot 2025-02-07 at 22 51 08" src="https://github.com/user-attachments/assets/fff2d40a-d4d2-41c7-b684-285cc571d7d6" />

pfrazee commented:
Yeah actually let's remove that in that case

estrattonbailey commented:
Discussed in Slack and decided to err on the side of increasing friction for following a user and then replying, if indeed the `followedBy` rule is in place.

@mozzius noting that since we show placeholder data from the cache here and get the `threadgate` from `getPostThread`, there is a slight flash of the Follow button rendering and then hiding.

I felt this was better than the alternative of not showing the Follow button and then showing it once the thread fully loaded, since that will affect the vast majority of posts, and the small regression here only affects posts with this threadgate enabled.

estrattonbailey commented:
Also noting that initially I had reverted the profile shadow use that was in place to ensure the reply button responded to a follow from the post thread. But I reverted that again so that post -> profile -> follow -> back to post enables replies. I think this would be confusing otherwise.

surfdude29 commented:
Sorry for jumping in here with one small point that just occurred to me: if you view a post that has the `followedBy` threadgate set on a feed (e.g. Discover or on someone's profile) on native and you don't follow the author, would the follow option be unavailable when tapping on the author's avatar, too? It's not quite as big and obvious a CTA, of course.

mozzius commented:
I don't mind it being present on the hover popover etc - I think that's fine. the big CTA was enough

mozzius reviewed (approved):
Button flash is fine, not too bad
